### PR TITLE
Improve jQuery code

### PR DESCRIPTION
**Pull request by jordisala1991**
Avoid some deprecations and use arrow functions as much as possible.
Try to avoid using this, and instead use the parameters passed to the
function.

## Subject

I am targeting this branch, because this is where it can be applied.

Part of #7049 

I think it is pedantic since it does not fix any issue, the js deprecations are not showin anywhere. (and probably they are deprecated on 3.0 of jQuery not on 2.0)

### CODE REVIEW

File: assets/js/Admin.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ const Admin = {
     }
   },
   read_config() {
-    const data = $('[data-sonata-admin]').data('sonata-admin');
+    const data = jQuery('[data-sonata-admin]').data('sonata-admin');
```
Comment (jordisala1991) [inline assets/js/Admin.js:51]: Some consistency here, try to use always jQuery so it can be easily identified

File: assets/js/Admin.js [hunk 2]
```diff
@@ -94,8 +94,8 @@ const Admin = {
     if (Admin.get_config('USE_SELECT2')) {
       Admin.log('[core|setup_select2] configure Select2 on', subject);
 
-      jQuery('select:not([data-sonata-select2="false"])', subject).each(function eachSetupSelect2() {
-        const select = jQuery(this);
+      jQuery('select:not([data-sonata-select2="false"])', subject).each((index, element) => {
```
Comment (jordisala1991) [inline assets/js/Admin.js:97]: arrow functions preferred over normal functions

File: assets/js/Admin.js [hunk 3]
```diff
@@ -142,15 +142,10 @@ const Admin = {
       Admin.log('[core|setup_icheck] configure iCheck on', subject);
 
       const inputs = jQuery('input[type="checkbox"]:not(label.btn > input, [data-sonata-icheck="false"]), input[type="radio"]:not(label.btn > input, [data-sonata-icheck="false"])', subject);
-      inputs
-        .iCheck({
-          checkboxClass: 'icheckbox_square-blue',
-          radioClass: 'iradio_square-blue',
-        })
-        // See https://github.com/fronteed/iCheck/issues/244
```
Comment (jordisala1991) [inline assets/js/Admin.js:150]: This issue looks like it was solved some time ago

File: assets/js/Admin.js [hunk 4]
```diff
@@ -190,16 +185,16 @@ const Admin = {
           const isChecked = jQuery(`tbody input[type="checkbox"]:nth(${currentIndex})`, subject).prop('checked');
 
           // Check all checkbox between previous and current one clicked
-          jQuery('tbody input[type="checkbox"]', subject).each((i, e) => {
```
Comment (jordisala1991) [inline assets/js/Admin.js:193]: don't like 1 char var name

**Review by VincentLanglet — APPROVED** [review]
Any config we should add to eslint to promote arrow function (or other things) ?

**Comment by jordisala1991** [comment]
I think the default airbnb config is just fine. 

The thing with arrow function is that sometimes (when you use this) you can't use arrow functions. But since the each function (and most of the callbacks from jquery already provide a way to acces "this" via function parameter, we should really use that and be able to throw an arrow function. This way we don't end up with a lot of "this" that we don't know what  they are.

**Pull request merged**